src/flb_network.c

```diff
@@ -259,18 +264,15 @@ static int net_connect_sync(int fd, const struct sockaddr *addr, socklen_t addrl
                   fd, host, port);
 
         /*
-         * Prepare a timeout using select(2): we could use our own
+         * Prepare a timeout using poll(2): we could use our own
          * event loop mechanism for this, but it will require an
-         * extra file descriptor, the select(2) call is straightforward
+         * extra file descriptor, the poll(2) call is straightforward
          * for this use case.
          */
-        FD_ZERO(&wait_set);
-        FD_SET(fd, &wait_set);
 
-        /* Wait 'connect_timeout' seconds for an event */
-        timeout.tv_sec = connect_timeout;
-        timeout.tv_usec = 0;
-        ret = select(fd + 1, NULL, &wait_set, NULL, &timeout);
+        pfd_read.fd = fd + 1;
```

Comment:
Sorry for the late response, but shouldn't this be just `fd` instead of `fd + 1`?

Another thing is that the `select` waited for the fd to be writable, while the `poll` waits for it to be readable, not sure it makes any difference in this specific scenario though.

Reply:
right!
just `pfd_read.fd = fd;`

Reply:
Just to understand. Why we had a +1 and no more ?

Reply:
Description of the first argument to `select`:
```
This argument should be set to the highest-numbered file
              descriptor in any of the three sets, plus 1.
```

Reply:
Thanks for the answer.